proposals/pattern-variables.md

```diff
@@ -43,18 +43,23 @@
 	}
 	```
 
-- Also relax single-declaration rules across the entire declaring scope as long as each variable is assigned once.
+- Also relax single-declaration rules within expression boundaries as long as each variable is assigned once.
 	```cs
 	if (e is C c || e is Wrapper { Prop: C c }) ;
-	if (e is C c) ; else if (e is Wrapper { Prop: C c }) ;	
 	if (b ? e is C c : e is Warpper { Prop: C c }) ;
 	```
 	
 ## Detailed design
 
 ### Variable redeclaration
 
-- Pattern variables are allowed to be redeclared in the same scope if not already definitely assigned. These names can possibly reference either of variables based on the result of the pattern-matching at runtime.
+- Pattern variables are allowed to be redeclared in the following locations if not already definitely assigned:
+	- Within case labels for the same switch section (includes `when` clauses)
+	- Within a single expression (includes `is` expressions)
+	- Within a single pattern (includes `switch` expression arms)
+	- Across top-level condition expressions within a single `if` statement (includes `else if`)
```

Comment:
>Within a single expression (includes `is` expressions)

:memo: This should exclude `(e is bool x || e switch { bool x => x })` because there's a nested scope within switch.

>Across top-level condition expressions within a single `if` statement 

:memo: This should exclude `if (e is not int x) if (e is int x)` even though embedded statements won't introduce a new scope.